### PR TITLE
Correctly handle special characters when encoding domain names

### PR DESCRIPTION
**Pull request by leofeyer**

### CODE REVIEW

File: core-bundle/src/Resources/contao/library/Contao/Idna.php [hunk 1]
```diff
@@ -38,7 +38,7 @@ public static function encode($strDomain)
 			return '';
 		}
 
-		if (($encoded = idn_to_ascii($strDomain)) === false)
+		if (($encoded = idn_to_ascii($strDomain, IDNA_NONTRANSITIONAL_TO_ASCII)) === false)
```
Comment (ausi) [inline core-bundle/src/Resources/contao/library/Contao/Idna.php:41]: Some issues from other projects indicate that this constant might not always exist, should we a `defined()` check?

Comment (leofeyer) [reply core-bundle/src/Resources/contao/library/Contao/Idna.php:41]: Yes. Or can we use the value of the constant instead? It should be `16`, right?

Comment (ausi) [reply core-bundle/src/Resources/contao/library/Contao/Idna.php:41]:
```suggestion
		if (($encoded = idn_to_ascii($strDomain, defined(IDNA_NONTRANSITIONAL_TO_ASCII) ? IDNA_NONTRANSITIONAL_TO_ASCII : 16)) === false)
```

Comment (leofeyer) [reply core-bundle/src/Resources/contao/library/Contao/Idna.php:41]: I just realized that the Symfony Intl polyfill will always define the constant, so I guess we do not need the `defined()` check after all: https://github.com/symfony/polyfill-intl-idn/blob/639084e360537a19f9ee352433b84ce831f3d2da/bootstrap.php#L79-L81

Comment (ausi) [reply core-bundle/src/Resources/contao/library/Contao/Idna.php:41]: Reading more detailed it seems that the missing constants problem only exsisted in older PHP versions, so we should be fine here I think.

**Review by ausi — APPROVED** [review]

**Pull request merged**